include/seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp

```diff
@@ -0,0 +1,474 @@
+                ++(*this);
+            }
+
+            /*!\name Comparison operators
+             * \{
+             */
+            //!\brief Returns `true` if the host iterator reached the end, `false` otherwise.
+            friend constexpr bool operator==(iterator_type const & lhs, sentinel const & rhs) noexcept
+            {
+                return lhs.host_iter == rhs;
+            }
+
+            //!\copydoc operator==
+            friend constexpr bool operator==(sentinel const & lhs, iterator_type const & rhs) noexcept
+            {
+                return rhs == lhs;
+            }
+
+            //!\brief Returns `true` if the host iterator did not reach the end, `false` otherwise.
+            friend constexpr bool operator!=(iterator_type const & lhs, sentinel const & rhs) noexcept
+            {
+                return !(lhs == rhs);
+            }
+
+            //!\copydoc operator!=
+            friend constexpr bool operator!=(sentinel const & lhs, iterator_type const & rhs) noexcept
+            {
+                return rhs != lhs;
+            }
+            //!\}
+
+        private:
+            //!\brief Pointer to the underlying alignment-column.
+            alignment_column_type * host_ptr{nullptr};
+            //!\brief Wrapped iterator over aliased matrix column.
+            std::ranges::iterator_t<view_type> host_iter{};
+        }; // class iterator_type
+
+    public:
+        /*!\name Constructors, destructor and assignment
+         * \{
+         */
+        constexpr alignment_column_type() = default; //!< Defaulted.
+        constexpr alignment_column_type(alignment_column_type const &) = default; //!< Defaulted.
+        constexpr alignment_column_type(alignment_column_type &&) = default; //!< Defaulted.
+        constexpr alignment_column_type & operator=(alignment_column_type const &) = default; //!< Defaulted.
+        constexpr alignment_column_type & operator=(alignment_column_type &&) = default; //!< Defaulted.
+        ~alignment_column_type() = default; //!< Defaulted.
+
+        /*!\brief Constructs from the derived type.
+         * \param[in] me  An instance of the derived type.
+         * \param[in] ref The underlying view referencing the current column in the matrix.
+         *
+         * \details
+         *
+         * This constructor is called by the derived type when invoking the function `initialise_column`.
+         */
+        constexpr alignment_column_type(derived_t & me, view_type ref) :
+            ref{std::move(ref)},
+            me_ptr{&me}
+        {}
+        //!\}
+
+        /*!\name Iterators
+         * \brief This column type is not const-iterable.
+         * \{
+         */
+        //!\brief Returns an iterator to the begin of the column.
+        constexpr auto begin() noexcept
+        {
+            assert(me_ptr != nullptr);
+            return iterator_type{*this};
+        }
+
+        //!\brief Deleted begin for const-qualified alignment-columns.
```

Comment:
because there is currently no need for this and allowing cons-iterability does add a lot more complexity to the iterators and the other classes. This range interface implemented here is strongly coupled with the algorithm and it doesn't fit to add something that the algorithm does not need. I don't see another place where this might be used. 
I do add a comment though.

Reply:
I did not ask you to change it ;) I was just curious

Reply:
I know, I was merely describing my motivation why I didn't add it. I am sorry if you felt offended. That was not my intention. 👼